### PR TITLE
Use findtext function

### PR DESCRIPTION
**Pull request by smileprem**

### CODE REVIEW

**Comment by smileprem** [comment]
Please review after the MR https://github.com/mercedes-benz/odxtools/pull/50 is merged.

**Comment by andlaus** [comment]
thanks for the PR. I've just merged #50. With that out of the way, I think the the 30-odd "old" patches are getting a bit unwieldy. can you remove them? AFAICS, this basically comes down to
```bash
git fetch origin
git rebase -i origin/main
# delete all lines except the one concerning the last commit
```

**Comment by smileprem** [comment]
Hi @andlaus - I fixed the merge conflicts. Please review

File: odxtools/multiplexer.py [hunk 1]
```diff
@@ -149,16 +149,8 @@ def __repr__(self) -> str:
...
-        int(et_element.find("BYTE-POSITION").text)
-        if et_element.find("BYTE-POSITION") is not None
-        else None
-    )
-    bit_position = (
-        int(et_element.find("BIT-POSITION").text)
-        if et_element.find("BIT-POSITION") is not None
-        else 0
-    )
+    byte_position = int(et_element.findtext("BYTE-POSITION"))
```
Comment (kayoub5) [inline odxtools/multiplexer.py:152]: byte position is optional, int(None) will fail

File: odxtools/multiplexer.py [hunk 2]
```diff
@@ -149,16 +149,8 @@ def __repr__(self) -> str:
...
+    bit_position = int(et_element.findtext("BIT-POSITION", 0))
```
Comment (kayoub5) [inline odxtools/multiplexer.py:153]: bit position is optional, int(None) will fail

File: odxtools/multiplexer.py [hunk 3]
```diff
@@ -208,15 +200,11 @@ def read_case_from_odx(et_element):
...
+    short_name = et_element.findtext("SHORT-NAME")
     long_name = et_element.find("LONG-NAME")
     if long_name is not None:
         long_name = long_name.text
-    byte_position = (
-        int(et_element.find("BYTE-POSITION").text)
-        if et_element.find("BYTE-POSITION") is not None
-        else None
-    )
+    byte_position = int(et_element.findtext("BYTE-POSITION"))
```
Comment (kayoub5) [inline odxtools/multiplexer.py:207]: same here

File: odxtools/parameters/readparameter.py [hunk 4]
```diff
@@ -28,10 +28,8 @@ def read_parameter_from_odx(et_element):
 
     semantic = et_element.get("SEMANTIC")
 
-    byte_position = int(et_element.find(
-        "BYTE-POSITION").text) if et_element.find("BYTE-POSITION") is not None else None
-    bit_position = int(et_element.find(
-        "BIT-POSITION").text) if et_element.find("BIT-POSITION") is not None else 0
+    byte_position = int(et_element.findtext("BYTE-POSITION"))
```
Comment (kayoub5) [inline odxtools/parameters/readparameter.py:31]: same here

File: odxtools/multiplexer.py (hunk 2 above)
Comment (andlaus) [reply odxtools/multiplexer.py:153]: while new solution will work, it is IMO nicer if a function call always returns objects of the same type, i.e.,
```suggestion
    bit_position = int(et_element.findtext("BIT-POSITION", "0"))
```

a slightly more severe issue with this: if the input file does not specify a bit position, it is impossible to detect that fact anymore, i.e., all output files written must either skip the tag for bit_position == 0 or the tag must always be written.

Comment (andlaus) [reply odxtools/multiplexer.py:153]: adding a `findint(et_element, location, default=None)` utility function analogous to `findtext()` would be a workable solution. I'm just not sure if that is a good idea, though...

**Review by andlaus — CHANGES_REQUESTED** [review]
in principle this is good, and I appreciate the PR. That said, the PR is probably a bit over-eager as constructs like `foo = int(et_elem.findtext("BAR", "0"))` will slightly change the contents of files written by odxtools.

also, please get rid of the  commits containing already merged functionality (i.e., all but the last two as far as I can see)

**Comment by smileprem** [comment]
Hi @andlaus @kayoub5 - Thanks for the suggestion. This MR is supposed to be a simple search and replace of find() with findtext(). Along the way, I made minor tweaks, which of course would produce un-intended consequences.

Shall we please scope this MR to search and replace only?. We can implement the other improvements in subsequent MRs. I'll fix all the MR comments today. Please let me know, if this is OK?.

File: odxtools/multiplexer.py (hunk 1 above)
Comment (smileprem) [reply odxtools/multiplexer.py]: fixed

File: odxtools/multiplexer.py (hunk 2 above)
Comment (smileprem) [reply odxtools/multiplexer.py]: reverted the code

File: odxtools/multiplexer.py (hunk 3 above)
Comment (smileprem) [reply odxtools/multiplexer.py]: fixed

File: odxtools/parameters/readparameter.py (hunk 4 above)
Comment (smileprem) [reply odxtools/parameters/readparameter.py]: fixed

**Comment by smileprem** [comment]
I'm used to git merge. I think, I messed up the git rebase. Let me close this and open another MR. Sorry about the trouble.

**Pull request closed without merging**